lib/models/container-store.js

```diff
@@ -75,10 +99,57 @@ ContainerStore.prototype.listContainers = function () {
           Id: container.Id,
           Image: container.Image,
           Created: container.Created,
-          Names: Names
+          Names: Names,
+          Labels: keypather.get(container, 'Config.Labels'),
+          State: container.State
         }
       }.bind(this))
     })
+    .then(function (containers) {
+      return ContainerStore._runFilters(containers, filters)
+    })
+}
+
+ContainerStore._runFilters = function (containers, filters) {
+  Object.keys(filters).forEach(function (filter) {
+    switch (filter) {
+      case 'label':
+        var key = Object.keys(filters[filter])[0]
+        var value = filters[filter][key]
+        containers = containers.filter(filterLabel(key, value))
+        break
+      case 'status':
+        var running = filters[filter] === 'running'
+        containers = containers.filter(filterStatus(running))
+        break
+    }
+  })
+  return containers
+
+  function filterLabel (key, value) {
+    return function (c) {
+      if (!c.Labels) { return false }
+      var containerValue = c.Labels[key]
+      if (containerValue === '' && value === '') { return true }
+      if (!containerValue || containerValue !== value) { return false }
+      return true
+    }
+  }
+  function filterStatus (status) {
+    return function (c) { return c.State.Running === status }
```

Comment:
I'm not sure what you're getting at. I do mention in the description above that `running` is the only supported filter at the moment. I'm sure I can write out the rest of the states eventually, but I don't see a need for it at the very moment